subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/cntr.c

```diff
@@ -47,17 +116,89 @@ uint32_t cntr_stop(void)
 		return 1;
 	}
 
+#if defined(CONFIG_BT_CTLR_NRF_GRTC)
+	/* TODO: if we own and stop GRTC, implement stop here */
+#else /* !CONFIG_BT_CTLR_NRF_GRTC */
 	nrf_rtc_task_trigger(NRF_RTC, NRF_RTC_TASK_STOP);
+#endif /* !CONFIG_BT_CTLR_NRF_GRTC */
 
 	return 0;
 }
 
 uint32_t cntr_cnt_get(void)
 {
+#if defined(CONFIG_BT_CTLR_NRF_GRTC)
+	uint32_t cntr_l, cntr_h, cntr_h_overflow;
+
+	/* NOTE: For a 32-bit implementation, L value is read after H
+	 *       to avoid another L value after SYSCOUNTER gets ready.
+	 *       If both H and L values are desired, then swap the order and
+	 *       ensure that L value does not change when H value is read.
+	 */
+	do {
+		cntr_h = nrf_grtc_sys_counter_high_get(NRF_GRTC);
+		cntr_l = nrf_grtc_sys_counter_low_get(NRF_GRTC);
+		cntr_h_overflow = nrf_grtc_sys_counter_high_get(NRF_GRTC);
+	} while ((cntr_h & GRTC_SYSCOUNTER_SYSCOUNTERH_BUSY_Msk) ||
+		 (cntr_h_overflow & GRTC_SYSCOUNTER_SYSCOUNTERH_OVERFLOW_Msk));
+
+	return cntr_l;
+#else /* !CONFIG_BT_CTLR_NRF_GRTC */
 	return nrf_rtc_counter_get(NRF_RTC);
+#endif /* !CONFIG_BT_CTLR_NRF_GRTC */
 }
 
 void cntr_cmp_set(uint8_t cmp, uint32_t value)
 {
+#if defined(CONFIG_BT_CTLR_NRF_GRTC)
+	uint32_t cntr_l, cntr_h, cntr_h_overflow, stale;
+
+	/* NOTE: We are going to use TASKS_CAPTURE to read current
```

Comment:
I don't follow why you are doing this.
You should be able to just read SYSCOUNTER to get the current GRTC counter value.
Why do you use the capture task instead?

Reply:
I just meant that it seems the use of the task to capture could have been instead replaced by using nrfy_grtc_sys_counter_get()